### ISSUE TITLE
Multiple Methods per Client

### ISSUE THREAD

**Issue by aowen14** [autonomi-ai/nos#493]
I have a sentence embedding model. Sometimes, I want a sentence similarity/cosine similarity score calculated, sometimes I want the encoding itself. I'd like to package these both up in the same client so that it can run on the server. That would mean 2 different entrypoints. "Default_method" is easy enough to use.

However, when defining a second method, it isn't callable via the test suite in the same manner as the default method. For example, if the model file has declarations of "embed", and "sentence_similarity", where default_method = embed, then calling model.sentence_similarity says it has no attribute "sentence_similarity", and from following the docs, model._call_(_method="sentence_similarity") doesn't work because it says the model has no attribute "_call_".

I've tried going through the docs to fix, but to no avail.

Any tips here? Happy to post a PR to the example repo to illustrate further

**Comment by spillai** [comment]
Can you try `model(..., _method="sentence_similarity")` similar to [this](https://github.com/autonomi-ai/nos/blob/cda4cf3c3d94ddc2618c623dff4e86c8f90796bf/tests/client/grpc/test_grpc_client_integration.py#L183)?

You probably hit that error because the `call` method is not implemented, it's the `__call__` method that you can use via `model(...)`.

Technically, both methods should be callable with the client `Module` interface. i.e. 
```python

model = client.Module("custom-model")

model(...)  # calls the default_method

# preferred sub-method calls
model.method1(...)  # explicitly calls method1
model.method2(...)  # explicitly calls method2

# less preferred sub-method calls
model(..., _method="method1") # calls method1
model(..., _method="method2") # calls method2
```

If you can provide a reproducable example, we'd love to take a look why this might be happening.

**Comment by spillai** [comment]
Resolved in #492. We replaced the `Hub.register()` method that registers a specific method with [`ModelSpec.from_cls`](https://github.com/autonomi-ai/nos/commit/1de45398f88dc08d114514705b431c0dd2a33dd0#diff-4fc63c133791b838182502f271e6b1e9dd1bc670547d42db37f0a11a741abcb4R311) that wraps the entire class and its submethods

**Comment by spillai** [comment]
Resolved above
